lib/ecl/ecl_file.c

```diff
@@ -542,10 +542,8 @@ static bool ecl_file_scan( ecl_file_type * ecl_file ) {
 
     ecl_kw_free( work_kw );
   }
-  if (scan_ok)
-    ecl_file_view_make_index( ecl_file->global_view );
-
-  return scan_ok;
+  ecl_file_view_make_index( ecl_file->global_view );
+  return (scan_ok || ignore_errors);
```

Comment:
It's maybe unnecessary that this method handles the `ignore_errors` flag. Whoever is calling this method is aware of `ignore_errors` so it can do
```
void whatever( ..., bool ignore_errors) {
  ....
   bool result = ecl_file_scan( .... )
   if(result || ignore_errors)
      ...
}
```

Reply:
Yes - that is a good idea.